lib/v2/datasource/remote/model/profile_model.dart

```diff
@@ -55,17 +55,45 @@ class ProfileModel {
 
   factory ProfileModel.fromJson(Map<String, dynamic> json) {
     return ProfileModel(
-      account: json['account'],
-      status: json['status'],
-      type: json['type'],
-      nickname: json['nickname'],
-      image: json['image'],
-      story: json['story'],
-      roles: json['roles'],
-      skills: json['skills'],
-      interests: json['interests'],
-      reputation: json['reputation'],
-      timestamp: json['timestamp'],
+      account: hasValue<String>(_account, json),
+      status: hasEmptyValue<String>(_status, json),
+      type: hasEmptyValue<String>(_type, json),
+      nickname: hasEmptyValue<String>(_nickname, json),
+      image: hasEmptyValue<String>(_image, json),
+      story: hasEmptyValue<String>(_story, json),
+      roles: hasEmptyValue<String>(_roles, json),
+      skills: hasEmptyValue<String>(_skills, json),
+      interests: hasEmptyValue<String>(_interests, json),
+      reputation: hasEmptyValue<int>(_reputation, json),
+      timestamp: hasValue<int>(_timestamp, json),
     );
   }
+
+  static final String _account = 'account';
+  static final String _status = 'status';
+  static final String _type = 'type';
+  static final String _nickname = 'nickname';
+  static final String _image = 'image';
+  static final String _story = 'story';
+  static final String _roles = 'roles';
+  static final String _skills = 'skills';
+  static final String _interests = 'interests';
+  static final String _reputation = 'reputation';
+  static final String _timestamp = 'timestamp';
+}
+
+T hasValue<T>(String mapKey, Map<String, dynamic> json) {
```

Comment:
Do you think we might need these functions in other models? 
If so, can we move these functions to a place thats shared and other models can access it?

Reply:
Very problaby we need in others

Reply:
Shouls I do it now or later and in what folder and file name?